packages/eslint-plugin-svelte/tests/fixtures/rules/prefer-svelte-reactivity/invalid/date01-input.svelte

```diff
@@ -0,0 +1,5 @@
+<script>
+  const variable = new Date(8.64e15);
+</script>
+
+{variable}
```

Comment:
I don’t think this needs to be reported. If there’s no reassignment, it shouldn’t need to be reactive.

Reply:
It's not about reassignment, it's about internal changes. I'd prefer to be this overly-broad. The alternative is tracking calls to every method that modifies the object (I count 16 on `Date` alone) or writing to any property.

On top of that, once something is exported, it should be reported as well (mostly relevant for `.svelte.ts` files)

I am not sure it's worth it, what do you think?

Reply:
How about tracking instances and checking mutable function calls?
For example, tracking `new Set()` variables and checking `add()` calls.

For example, I might use:

```js
const ALLOWED_VALUES = new Set(["a", "b"]); // Not reported
function isAllowedValue(v) {
  return ALLOWED_VALUES.has(v);
}

const usedValues ​​= new Set();
function process(v) {
  if (usedValues.has(v)) return;
  usedValues.add(v); // Reported
  // ...
}
```

Reply:
Yeah, that's probably a better way to do it...